tests/cosim/emulator/src/tl_interface.h

```diff
@@ -0,0 +1,42 @@
+#include <VV.h>
+
+namespace TlOpcode {
+  constexpr int
+      AcquireBlock = 6,
+      Get = 4,
+      AccessAckData = 1,
+      PutFullData = 0,
+      PutPartialData = 1,
+      AccessAck = 4;
+}
+
+// the following macro helps us to access tl interface by dynamic index
+#define TL_INTERFACE(type, name)                      \
+  [[nodiscard]] inline type &get_tl_##name(VV &top) { \
+    return top.memory_0_##name;                       \
```

Comment:
I don't this using `.` from verilator is a good idea. it is hardcoded. Please switch to VPI or DPI instead.